packages/flutter_tools/lib/src/web/compiler_config.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+abstract class WebCompilerConfig {
+  const WebCompilerConfig();
+
+  /// Returns `true` if `this` represents configuration for the Wasm compiler.
+  ///
+  /// Otherwise, `false`–represents the JavaScript compiler.
+  bool get isWasm;
+
+  Map<String, String> toBuildSystemEnvironment();
+}
+
+/// Configuration for the Javascript compiler.
```

Comment:
Throughout this file, would it make sense to substitute "dart2js" for "The JavaScript compiler", as it is more specific?